### ISSUE TITLE
LSTM model compiles very slowly

### ISSUE THREAD

**Issue by BadrYoubiIdrissi** [pytorch/xla#1027]
I'm trying to train a language model on TPU. I've written some code that took way too long to compile the model for XLA. Here's some dummy code that reproduces my issue. I haven't come across any examples implementing RNNs for NLP in the test cases so maybe I'm doing something wrong? 
```
import torch
import torch_xla
import torch.nn as nn
import time
import model

import torch_xla_py.xla_model as xm
import torch_xla_py.data_parallel as dp

class Loader:
    def __iter__(self):
        return self
    def __next__(self):
        return torch.rand(100, 100, 100)

m = nn.LSTM(100,100)
devices = xm.get_xla_supported_devices()
m = dp.DataParallel(m, device_ids=devices, batchdim=1)

def train_loop(model, loader, device, context):
    h = torch.rand(1, 100, 100).to(device), torch.rand(1, 100, 100).to(device)
    t = 0
    for _, e in loader:
        tic = time.time()
        out = model(e, h)
        toc = time.time()
        print(toc-tic)

m(train_loop, Loader())
```

I've also dumped XLA logs using `export XLA_FLAGS=--xla_dump_to=/tmp/xlalogs` and it gave the following logs : 

[module_0000.after_optimizations.txt](https://github.com/pytorch/xla/files/3603880/module_0000.after_optimizations.txt)
[module_0000.after_optimizations-buffer-assignment.txt](https://github.com/pytorch/xla/files/3603881/module_0000.after_optimizations-buffer-assignment.txt)
[module_0000.before_optimizations.txt](https://github.com/pytorch/xla/files/3603882/module_0000.before_optimizations.txt)
[module_0000.ir-no-opt.txt](https://github.com/pytorch/xla/files/3603900/module_0000.ir-no-opt.txt)
[module_0000.ir-no-opt-noconst.txt](https://github.com/pytorch/xla/files/3603901/module_0000.ir-no-opt-noconst.txt)
[module_0000.txt](https://github.com/pytorch/xla/files/3603902/module_0000.txt)

**Comment by BadrYoubiIdrissi** [comment]
I finally ran some tests on a real TPU and indeed compilation time is good. The same model ran on TPU is approximately 3 times faster (when normalizing with batch size*sequence length) than a GTX 1080. However the TPU's usage is very low (5% to 10%). How could I identify operations that are running slowly? (Maybe those processed on CPU). CPU usage is not at 100% either and Network usage isn't exceptionally large so I'm guessing it's not the data loading that is bottlenecking(?)

**Comment by dlibenzi** [comment]
The metrics report can give some insights:
```Python
print(torch_xla._XLAC._xla_metrics_report())
```

**Comment by dlibenzi** [comment]
I am closing this, but if there is further investigation feel free to re-open.
